code-examples/missingNumber/src/MissingNumber.java

```diff
@@ -0,0 +1,33 @@
+import java.util.stream.IntStream;
+
+class MissingNumber {
+    public static void main(String[] args) {
+    }
+
+
+    int[] createArray(int range, int randomNumber) {
+        int[] arrStart = IntStream.range(1, randomNumber).toArray();
+        int[] arrEnd = IntStream.range(randomNumber + 1, range + 1).toArray();
+        int[] arr = new int[arrStart.length + arrEnd.length];
+        System.arraycopy(arrStart, 0, arr, 0, arrStart.length);
+        System.arraycopy(arrEnd, 0, arr, arrStart.length, arrEnd.length);
+        return arr;
+    }
+
+    int search(int[] arr, int lowI, int highI) {
```

Comment:
This is the method that does the real work: I didn't know whether or not to refactor a little further - or to make it more clear/verbose as to what the method does so that someone else can understand it.